apps/src/code-studio/header.js

```diff
@@ -173,21 +190,72 @@ function shareProject() {
 
     ReactDOM.render(
       <Provider store={getStore()}>
-        <ShareDialog
+        <ConnectedShareDialog
           i18n={i18n}
           icon={appOptions.skin.staticAvatar}
           shareUrl={shareUrl}
           isAbusive={dashboard.project.exceedsAbuseThreshold()}
+          isOwner={dashboard.project.isOwner()}
+          isPublished={dashboard.project.isPublished()}
           channelId={dashboard.project.getCurrentId()}
           appType={appType}
           onClickPopup={popupWindow}
           // TODO: Can I not proliferate the use of global references to Applab somehow?
           onClickExport={window.Applab ? window.Applab.exportApp : null}
+          onShowPublishDialog={showPublishDialog}
+          onUnpublish={unpublishProject}
           canShareSocial={canShareSocial}
         />
       </Provider>,
       dialogDom
     );
+
+    getStore().dispatch(showShareDialog());
+  });
+}
+
+const ConnectedPublishDialog = connect(state => ({
+  isOpen: state.header.publishDialog.isOpen,
+  isPublishPending: state.header.publishDialog.isPublishPending,
+  projectId: state.header.publishDialog.projectId,
+  projectType: state.header.publishDialog.projectType,
+}), dispatch => ({
+  onClose: () => dispatch(hidePublishDialog()),
+}))(PublishDialog);
```

Comment:
The problem is that there are two connected versions of ConnectedPublishDialog. If we can figure out how to pass a subtree of the store to this component, then we might be able to extract and share the `connect` logic.